lib/tapioca/dsl/helpers/graphql_type_helper.rb

```diff
@@ -47,8 +47,6 @@ def type_for(argument)
             end
           when GraphQL::Schema::InputObject.singleton_class
             type_for_constant(unwrapped_type)
-          when GraphQL::Schema::NonNull.singleton_class
-            type_for(unwrapped_type.of_type)
```

Comment:
With the Sorbet bump, I was warned that `GraphQL::Schema::NonNull.of_type` doesn't exist, which I verified with 

```
irb(main):008:0> GraphQL::Schema::NonNull.of_type
(irb):8:in `<main>': undefined method `of_type' for GraphQL::Schema::NonNull:Class (NoMethodError)
```

Also, this branch is never entered when running the tests. So I think we can remove it.

cc @paracycle

Reply:
Yes, but it's an instance method and the `unwrapped_type` here should be the `GraphQL::Schema::NonNull` class (type) itself.

Reply:
Did the method never exist or was it removed because of the GraphQL upgrade? This wasn't failing typechecking before, so there's a chance this only fails in the latest version.

Reply:
I think it's worth understanding `NonNull` better and try to get a test case with it. I don't think the original PR tests it https://github.com/Shopify/tapioca/commit/72dde31efa4f76b0906c7d936c79f255f9296f3b. And then we'll know if returning `type_for(unwrapped_type)` or `T.untyped` is correct.

Reply:
AFAICT, it doesn't exist in `main` either.
![Screenshot 2023-10-17 at 18 40 32](https://github.com/Shopify/tapioca/assets/5079556/9af17253-bf90-4b63-ac02-a8fe40ad7738)

It wasn't detected because even in the branch Sorbet still treated it as `untyped`. 

![Screenshot 2023-10-17 at 18 45 58](https://github.com/Shopify/tapioca/assets/5079556/7be8913f-d1ca-4717-8212-ecae9df25436)

But now Sorbet is able to detect the type of `unwrapped_type` in the branch, it catches the error:

![Screenshot 2023-10-17 at 18 46 24](https://github.com/Shopify/tapioca/assets/5079556/58e33576-5eb6-42c4-86aa-b40795498cbf)